### ISSUE TITLE
[Multimedia collage] Product card missing focus ring

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#1158]
## Issue summary

When navigating through page content with the keyboard, the Multimedia collage product card does not feature a focus indicator. Sighted keyboard-only users may not know where the cursor has landed and lead to a frustrating user experience.

<details>
  <summary>Screenshots</summary>

  <img width="845" alt="Multimedia collage, product card is highlighted." src="https://user-images.githubusercontent.com/1392632/149024517-cf7b26b3-89dd-454f-8b34-13197f35b37a.png">

</details>

<!-- ### Current code

#### HTML

```html

``` -->

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. <kbd>Tab</kbd> through a Multimedia Collage section

## Behavior

### Expected

- Product card to feature a focus indicator.

### Actual

- Focus indicator is missing.

## Recommendation

Apply the focus ring styling to the `.card-wrapper:focus-within .card` CSS selector. This will display the focus ring on the card when the inner link receives focus.

### Recommended code

#### CSS

```css
.card-wrapper:focus-within .card {
  outline: 0.2rem solid rgba(var(--color-foreground),.5);
  outline-offset: 0.3rem;
  box-shadow: 0 0 0 0.3rem rgb(var(--color-background)),0 0 0.5rem 0.4rem rgba(var(--color-foreground),.3);
}
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.5.0
- **Component:** Multimedia collage
- **WCAG Principal:** Operable
- **WCAG SC:** [2.4.7 Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)
- **Severity:** Medium
- **Effort:** Low
